src/com/oltpbenchmark/util/ResultUploader.java

```diff
@@ -1,204 +1,110 @@
-import org.apache.http.entity.mime.content.FileBody;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClients;
-import org.apache.http.util.EntityUtils;
-import org.apache.log4j.Logger;
-
-import java.io.*;
+import org.apache.commons.configuration2.XMLConfiguration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.configuration2.io.FileHandler;
+
+import java.io.PrintStream;
 import java.util.Date;
-import java.util.List;
 import java.util.Map;
 import java.util.TimeZone;
 import java.util.TreeMap;
-import java.util.zip.GZIPOutputStream;
 
 public class ResultUploader {
-    private static final Logger LOG = Logger.getLogger(ResultUploader.class);
 
-    private static String[] IGNORE_CONF = {
-            "dbtype",
+    private static final String[] IGNORE_CONF = {
+            "type",
             "driver",
-            "DBUrl",
+            "url",
             "username",
-            "password",
-            "uploadCode",
-            "uploadUrl"
+            "password"
     };
 
-    private static String[] BENCHMARK_KEY_FIELD = {
+    private static final String[] BENCHMARK_KEY_FIELD = {
             "isolation",
             "scalefactor",
             "terminals"
     };
 
-    XMLConfiguration expConf;
-    Results results;
-    CommandLine argsLine;
-    DBCollector collector;
+    private final XMLConfiguration expConf;
+    private final DBParameterCollector collector;
+    private final Results results;
+    private final DatabaseType dbType;
+    private final String benchType;
 
-    String dbUrl;
-    DatabaseType dbType;
-    String username, password;
-    String benchType;
-//    int windowSize;
-    String uploadCode, uploadUrl;
-    String uploadHash;
 
     public ResultUploader(Results r, XMLConfiguration conf, CommandLine argsLine) {
         this.expConf = conf;
         this.results = r;
-        this.argsLine = argsLine;
-
-        dbUrl = expConf.getString("DBUrl");
-        dbType = DatabaseType.get(expConf.getString("dbtype"));
-        username = expConf.getString("username");
-        password = expConf.getString("password");
-        benchType = argsLine.getOptionValue("b");
-//        windowSize = 1;
-//        if (argsLine.hasOption("s")) {
-//        	windowSize = Integer.parseInt(argsLine.getOptionValue("s"));
-//        } else {
-//        	windowSize = 1;
-//        }
-        uploadCode = expConf.getString("uploadCode");
-        uploadUrl = expConf.getString("uploadUrl");
-        uploadHash = argsLine.getOptionValue("uploadHash");
-        uploadHash = uploadHash == null ? "" : uploadHash;
-
-        this.collector = DBCollector.createCollector(dbType, dbUrl, username, password);
-        assert(this.collector != null);
-    }
-    
-    public DBCollector getConfCollector() {
-        return (this.collector);
+        this.dbType = DatabaseType.valueOf(expConf.getString("type"));
```

Comment:
I don't know whether we actually even still use this code, but if we do, check breakage